src/gui/advancedsettings.cpp

```diff
@@ -619,6 +640,12 @@ void AdvancedSettings::loadAdvancedSettings()
     m_spinBoxSaveResumeDataInterval.setSuffix(tr(" min", " minutes"));
     m_spinBoxSaveResumeDataInterval.setSpecialValueText(tr("0 (disabled)"));
     addRow(SAVE_RESUME_DATA_INTERVAL, tr("Save resume data interval [0: disabled]", "How often the fastresume file is saved."), &m_spinBoxSaveResumeDataInterval);
+    // .torrent file size limit
+    m_spinBoxTorrentFileSizeLimit.setMinimum(1);
+    m_spinBoxTorrentFileSizeLimit.setMaximum(std::numeric_limits<int>::max() / 1024 / 1024);
```

Comment:
Wouldn't it be more efficient if use only one division, i.e. `val / (1024 * 1024)`?
And considering that we are obviously talking about powers of two, then maybe it's even worth using shifts? I.e. `val >> 20`?

Reply:
>Wouldn't it be more efficient if use only one division, i.e. val / (1024 * 1024)?

I won't worry too much about it. Once compiler optimization kicks in it will be optimized out to constant.
https://godbolt.org/z/1179dPP4n

>And considering that we are obviously talking about powers of two, then maybe it's even worth using shifts? I.e. val >> 20?

https://en.wikipedia.org/wiki/Strength_reduction
I wouldn't want to make the code obscure.